Framework/PythonInterface/mantid/utils/absorptioncorrutils.py

```diff
@@ -611,7 +679,31 @@ def confirmProps(props):
     # Make sure one is set before calling SetSample
     if material or geometry or environment:
         mantid.simpleapi.SetSampleFromLogs(
-            InputWorkspace=absName, Material=material, Geometry=geometry, Environment=environment, FindEnvironment=find_environment
+            InputWorkspace=absName,
+            Material=material,
+            Geometry=geometry,
+            ContainerGeometry=can_geometry,
+            ContainerMaterial=can_material,
+            Environment=environment,
+            FindEnvironment=find_environment,
         )
 
+    if beam_height != Property.EMPTY_DBL and not gauge_vol:
+        # If the gauge volume is not defined, use the beam height to define it,
+        # and we will be assuming a cylinder shape of the sample.
+        gauge_vol = """<cylinder id="shape">
+            <centre-of-bottom-base r="{0:4.2F}" t="90.0" p="270.0" />
+            <axis x="0.0" y="0.2" z="0.0" />
```

Comment:
I believe it's not always the case that the vertical is along the y-axis - I think you can get this from the IDF/instrument?

Reply:
Sorry about the delay on my side due to local contacting experiments. I am not sure what is the best way to obtain the axis direction. Could you please help provide some hints about how specifically we can do the checking? Thanks!

Reply:
No worries, you can access this from the instrument reference frame like so
https://github.com/mantidproject/mantid/blob/3306871804d5a749a7c0fb248d18f7eab83812ca/Testing/SystemTests/tests/framework/RelectometryInstrumentSignedThetaTest.py#L25-L30

Reply:
But if you know all the instruments using this script have the y-axis vertical then I'm happy to keep as it is!

Reply:
> But if you know all the instruments using this script have the y-axis vertical then I'm happy to keep as it is!

To be honest, I am not sure at this moment. I will check with @peterfpeterson and folks here to see. But if the implementation for incorporating the instrument definition is not so involved, I will do it anyhow. Will see and will let you know. Thanks a ton!